scanner/config/config_test.go

```diff
@@ -91,4 +91,102 @@ func Test_validate(t *testing.T) {
 		err := c.validate()
 		assert.ErrorContains(t, err, "grpc_listen_addr is empty")
 	})
+	t.Run("when indexer is invalid then error", func(t *testing.T) {
+		c := defaultConfiguration
+		c.Indexer.Database.ConnString = "force indexer to fail validate"
+		err := c.validate()
+		assert.ErrorContains(t, err, "indexer:")
+	})
+	t.Run("when matcher is invalid then error", func(t *testing.T) {
+		c := defaultConfiguration
+		c.Matcher.Database.ConnString = "force matcher to fail validate"
+		err := c.validate()
+		assert.ErrorContains(t, err, "matcher:")
+	})
+}
+
+func Test_IndexerConfig_validate(t *testing.T) {
+	t.Run("when disabled no error", func(t *testing.T) {
+		c := IndexerConfig{Enable: false, Database: Database{ConnString: "invalid conn string"}}
+		err := c.validate()
+		assert.NoError(t, err)
+	})
+	t.Run("when enabled then error", func(t *testing.T) {
+		c := IndexerConfig{Enable: true, Database: Database{ConnString: "invalid conn string"}}
+		err := c.validate()
+		assert.Error(t, err)
+	})
+}
+
+func Test_MatcherConfig_validate(t *testing.T) {
+	t.Run("when disabled no error", func(t *testing.T) {
+		c := MatcherConfig{Enable: false, Database: Database{ConnString: "invalid conn string"}}
+		err := c.validate()
+		assert.NoError(t, err)
+	})
+	t.Run("when enabled then error", func(t *testing.T) {
+		c := MatcherConfig{Enable: true, Database: Database{ConnString: "invalid conn string"}}
+		err := c.validate()
+		assert.Error(t, err)
+	})
+}
+
+func Test_Database_validate(t *testing.T) {
+	//	# Example DSN
+	//	user=jack password=secret host=pg.example.com port=5432 dbname=mydb sslmode=verify-ca pool_max_conns=10
+	//
+	//	# Example URL
+	//	postgres://jack:secret@pg.example.com:5432/mydb?sslmode=verify-ca&pool_max_conns=10
+	t.Run("when DNS then no error", func(t *testing.T) {
+		c := Database{ConnString: "user=jack password=secret host=pg.example.com port=5432 dbname=mydb sslmode=verify-ca pool_max_conns=10"}
+		err := c.validate()
+		assert.NoError(t, err)
+	})
+	t.Run("when URL then valid", func(t *testing.T) {
+		c := Database{ConnString: "postgres://jack:secret@pg.example.com:5432/mydb?sslmode=verify-ca&pool_max_conns=10"}
+		err := c.validate()
+		assert.ErrorContains(t, err, "URLs are not supported")
+	})
+	t.Run("when empty conn string then error", func(t *testing.T) {
+		c := Database{ConnString: ""}
+		err := c.validate()
+		assert.ErrorContains(t, err, "empty is not allowed")
+	})
+	t.Run("when conn string is not parsable then error", func(t *testing.T) {
+		c := Database{ConnString: "this is nothing meaningful"}
+		err := c.validate()
+		assert.ErrorContains(t, err, "cannot parse")
+	})
+
+	tempDir, err := os.MkdirTemp("", "Test_Database_validate")
```

Comment:
nit: you can use https://pkg.go.dev/testing#T.TempDir, which will also be automatically cleaned at the end of this test